llvm/include/llvm/Analysis/ConditionCacheUtil.h

```diff
@@ -0,0 +1,131 @@
+//===- llvm/Analysis/ConditionCacheUtil.h -----------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Shared by DomConditionCache and AssumptionCache. Holds common operation of
+// finding values potentially affected by an assumed/branched on condition.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_ANALYSIS_CONDITIONCACHEUTIL_H
+#define LLVM_ANALYSIS_CONDITIONCACHEUTIL_H
```

Comment:
Why does this code live inside a header?

Reply:
(If having it in a header is not necessary for compile-time reasons, then maybe consider putting this in ValueTracking? That way the logic for determining affected values and the logic that makes use of them is closer together. No strong feelings on this though.)

Reply:
in ValueTracking.cpp then just forward declare in AssumptionCache/DomConditionCache?

Reply:
Doesn't seem to be any extra overhead: https://llvm-compile-time-tracker.com/compare.php?from=d95e97609b15dc13a2d20dc9edb0257b742af86b&to=df14b4d584162ca73a8208b4055d05abea7f07d5&stat=instructions:u

moving to valuetracking